pinc/misc.inc

```diff
@@ -1272,6 +1272,42 @@ function factor_strings($strings)
     return [$left_common, $middles, $right_common];
 }
 
+//--------------------------------------------------------------------------
+
+/**
+ * Given a function and arguments that will generate data, attempt
+ * to load the data from memcached first and fall back to the function and
+ * cache the response. If memcache is not running, this will call the requested
+ * function and return its data without generating a notice/warning/error.
+ *
+ * If the requested function returns data that has localized strings, it's
+ * important that $key_salt = get_desired_language();
+ */
+function memoize_function($function, $args = [], $expire_from_now = 3600, $key_salt = "")
+{
+    // cache this connection for possible re-use during this page load
+    static $memcache = null;
+    if (!$memcache) {
+        $memcache = new Memcached();
+        $memcache->addServer('localhost', 11211);
```

Comment:
minor point, but should we make the port and server configurable similar to other dependencies?

Reply:
We probably should but I'm going to punt on that right now. The install docs talk about this being accessible on localhost so they at least reflect reality.